static/js/lib/ckeditor/plugins/util.test.ts

```diff
@@ -1,4 +1,10 @@
-import { unescapeStringQuotedWith, Shortcode, ShortcodeParam } from "./util"
+import {
+  unescapeStringQuotedWith,
+  Shortcode,
+  ShortcodeParam,
+  replaceShortcodes,
+  findNestedExpressions
+} from "./util"
```

Comment:
These frontend shortcode parsing utilities are now, IMO, pretty good. But if we need to do anything much more with the shortcodes (e.g., support opening/closing shortcodes in CKEditor) it may be worth using a more formal grammar / parsing approach. The backend is using PyParsing for this. A similar frontend library (that I have less experience with) is https://www.npmjs.com/package/nearley